client_test.go

```diff
@@ -280,6 +282,50 @@ func (s *ClientSuite) TestGetBasicAuth(c *C) {
 	c.Assert(pass, DeepEquals, "pass")
 }
 
+func (s *ClientSuite) TestCookies(c *C) {
+	var capturedRequestCookies []*http.Cookie
+	responseCookies := []*http.Cookie{
+		{
+			Name:  "session",
+			Value: "howdy",
+			Path:  "/",
+		},
+	}
+	srv := serveHandler(func(w http.ResponseWriter, r *http.Request) {
+		capturedRequestCookies = r.Cookies()
+		for _, c := range responseCookies {
+			http.SetCookie(w, c)
+		}
+	})
+	defer srv.Close()
+
+	jar, err := cookiejar.New(nil)
+	c.Assert(err, IsNil)
+	clt := newC(srv.URL, "v1", CookieJar(jar))
+
+	requestCookies := []*http.Cookie{
+		{
+			Name:  "hello",
+			Value: "here?",
+			Path:  "/",
+		},
+	}
+	u, err := url.Parse(srv.URL)
+	c.Assert(err, IsNil)
+	jar.SetCookies(u, requestCookies)
+
+	re, err := clt.Get(clt.Endpoint("test"), url.Values{})
+	c.Assert(err, IsNil)
+
+	c.Assert(len(capturedRequestCookies), Equals, len(requestCookies))
+	c.Assert(capturedRequestCookies[0].Name, DeepEquals, requestCookies[0].Name)
+	c.Assert(capturedRequestCookies[0].Value, DeepEquals, requestCookies[0].Value)
```

Comment:
These tests are too obscure due to verbosity. I think

```go
c.Assert(capturedRequestCookies, DeepEquals, requestCookies)
c.Assert(re.Cookies(), DeepEquals, responseCookies)
```
should be more readable while keeping the functional element. Although it might be problematic for arrays with more than one element (not the case here), but then they can be sorted and still asserted as a whole.